packages/mask/src/plugins/Pets/Services/rss3.ts

```diff
@@ -0,0 +1,69 @@
+import Web3 from 'web3'
+import RSS3 from 'rss3-next'
+import { isSameAddress } from '@masknet/web3-shared-evm'
+import { personalSign } from '../../../extension/background-script/EthereumService'
+import type { EssayRSSNode, PetMetaDB } from '../types'
+import { RSS3_APP } from '../../Avatar/constants'
+
+export async function createRSS3(address: string) {
+    return new RSS3({
+        endpoint: RSS3_APP,
+        address,
+        sign: async (message: string) => {
+            return personalSign(message, address)
+        },
+    })
+}
+
+const cache = new Map<string, Promise<PetMetaDB | undefined>>()
+
+export async function getCustomEssayFromRSS(address: string) {
+    let f = cache.get(address)
+    if (!f) {
+        f = _getCustomEssayFromRSS(address)
+        cache.set(address, f)
+    }
+    const essay = await f
+    return essay
+}
+
+async function _getCustomEssayFromRSS(address: string): Promise<PetMetaDB | undefined> {
+    const rss = await createRSS3(address)
```

Comment:
Do you need to create a new RSS3 instance every time? Do you need to close it in case it starts something like WebSockets that continuously connect to the internet?

Reply:
i think this code is work `const cache = new Map<string, Promise<PetMetaDB | undefined>>()`